src/Data/Nat/Divisibility/Core.agda

```diff
@@ -35,15 +35,55 @@ record _∣_ (m n : ℕ) : Set where
   constructor divides
   field quotient : ℕ
         equality : n ≡ quotient * m
-open _∣_ using (quotient) public
 
+  quotient≡0⇒n≡0 : quotient ≡ 0 → n ≡ 0
+  quotient≡0⇒n≡0 q≡0 = subst (λ q → n ≡ q * m) q≡0 equality
```

Comment:
Why don't you match on `refl` and let arithmetic just give you the proof? Or use an equational proof? `subst` seems like such a ill-suited hammer for this.

Reply:
Good idea! but *in the scope of the record*, matching on `refl` fails...
```agda
I'm not sure if there should be a case for the constructor refl,
because I get stuck when trying to solve the following unification
problems (inferred index ≟ expected index):
  quotient ≟ 0
when checking that the pattern refl has type quotient ≡ 0
```

Reply:
Then I might be tempted to prove a lemma in `Data.Nat.Properties` so that you can use it here?

But I wonder if the matching failure is an Agda bug.

Reply:
> Then I might be tempted to prove a lemma in `Data.Nat.Properties` so that you can use it here?

Re: lemma? I'm happy with the new proofs... are you?

> But I wonder if the matching failure is an Agda bug.

Re: bug? Not sure, but if you can reproduce it with an MWE, do please post an issue on the main `agda` tracker!
(I'm not sure if I can be bothered at this point... sorry!)

Reply:
When you can't match on `refl`, try `rewrite`...: which do you prefer?
```agda
  quotient≡0⇒n≡0 q≡0 rewrite equality | q≡0 = refl
```
etc.

Reply:
I guess I prefer `rewrite` over `subst`, but prefer explicit proofs over both. So weird that `rewrite` would work!

Reply:
As I understand it (but not very well... you'd think!?), `rewrite` *is* in fact encapsulating the abstraction over the proof of the equation, and then returning the `refl` instance of that type as the next subgoal/hole... which is morally what the previous `subst` proof was doing 'by hand'... 

... now, just like its grown-up friend `with`, that abstraction step might not be well-typed (hence your reasonable lament about the non-robustness of `rewrite`), but when it does work, I think FWIW that it does save a lot of too-much/too-unnecessary cognitive load on the user.

Reply:
@JacquesCarette my apologies, I think you are quite correct about preferring a proof in `Data.Nat.Properties`, not least because it reduces the module interface here. Incoming commit...